src/Avalonia.Controls.TreeDataGrid/Selection/TreeDataGridRowSelectionModel.cs

```diff
@@ -88,6 +88,23 @@ void ITreeDataGridSelectionInteraction.OnKeyDown(TreeDataGrid sender, KeyEventAr
                     {
                         e.Handled = MoveSelection(sender, direction.Value, shift, anchor);
                     }
+
+                    if (!e.Handled && direction == NavigationDirection.Left
+                        && anchor?.Rows is HierarchicalRows<T> hierarchicalRows && anchorRowIndex > 0)
+                    {
+                        var newIndex = hierarchicalRows.GetParentRowIndex(AnchorIndex);
```

Comment:
If we're at the root node then I think `newIndex` will be -1 here, and I think in that case the following instructions should maybe be skipped?

Reply:
I assume anchorRowIndex > 0 covers this case and in would not go inside of the if because of that.